### DISCUSSION TITLE
Not able to Use gpgpu::primitives::pixels::Luma8 and Luma8Norm;

### DISCUSSION THREAD

**Discussion by RahulDas-dev** [UpsettingBoy/gpgpu-rs#12] [Show and tell]
HI , 
Using gpgpu-rs, I am doing RGB to Gray Conversation, but I have not found **Luma8** or **Luma8Norm**. I see they were commented off in [v.0.2.0](https://github.com/UpsettingBoy/gpgpu-rs/blob/3beca2ce3c6b8c7b13c631cef61e98a1469561b1/src/primitives.rs#L144)

![image](https://user-images.githubusercontent.com/32712591/147733640-25bb0df1-2574-4e21-af2e-95faa2348622.png)

However, this two were present in [v.0.1.0](https://github.com/UpsettingBoy/gpgpu-rs/blob/46b7f7a452596fdde90f373c014fd86990e52b33/src/primitives.rs#L56)

I would appreciate any feedback on that.

**Comment by jeronimosg** [comment]
They are tricky to use in WGSL shaders. First they are loaded as a texture of a 32-bit primitive, thus making every 4 pixels on the CPU data a channel of a single pixel on the GPU (4 to 1). Then, the result cannot be properly saved because R32Uint storage textures or anything similar useful won't work on the shader. So they are difficult to work with and then challenging to save :)

I still trying bring them again, but for now, you have to upload the image as a Rgba8 one, duplicating or zeroing the extra data, and then only using 1 channel of the image/texture on the shader. Hope this helps!
